Classes/common/CDTEncryptionKey/CDTEncryptionKey.h

```diff
@@ -0,0 +1,26 @@
+//
+//  CDTEncryptionKey.h
+//
+//
+//  Created by Enrique de la Torre Fernandez on 20/02/2015.
+//
+//
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//  http://www.apache.org/licenses/LICENSE-2.0
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+
+@protocol CDTEncryptionKey <NSObject, NSCopying>
```

Comment:
You're right, the convention for a protocol like this is to use gerund form. `CDTEncryptionKeyRetrieving` is fine for me, it might sounds contrived but clearly  states the purpose of this protocol.